python/mlc_llm/serve/entrypoints/openai_entrypoints.py

```diff
@@ -403,17 +405,27 @@ async def request_chat_completion(
     # - Get the prompt from template, and encode to token ids.
     # - Check prompt length
     async_engine.record_event(request_id, event="start tokenization")
-    prompts = entrypoint_utils.process_prompts(
-        conv_template.as_prompt(), async_engine.tokenizer.encode
-    )
+
+    model_config_path = ServerContext.get_model_config_path(request.model)
+    image_embed_size = entrypoint_utils.get_image_embed_size(model_config_path)
+
+    if content_has_list:
+        prompts = entrypoint_utils.process_prompts(
+            conv_template.as_prompt_list(image_embed_size=image_embed_size),
+            async_engine.tokenizer.encode,
+        )
+    else:
+        prompts = entrypoint_utils.process_prompts(
+            conv_template.as_prompt(), async_engine.tokenizer.encode
+        )
```

Comment:
Just want to mark a future todo item. Let's unify `as_prompt_list` and `as_prompt` in the future. Could you help add a TODO in the code here?